packages/core/src/browser/fetchObservable.ts

```diff
@@ -93,25 +94,58 @@ function afterSend(
   responsePromise: Promise<Response>,
   startContext: FetchStartContext
 ) {
-  const reportFetch = (response: Response | Error) => {
-    const context = startContext as unknown as FetchCompleteContext
-    context.state = 'complete'
-    context.duration = elapsed(context.startClocks.timeStamp, timeStampNow())
-
+  const reportFetch = (context: FetchCompleteContext, response: Response | Error) => {
     if ('stack' in response || response instanceof Error) {
       context.status = 0
       context.isAborted = response instanceof DOMException && response.code === DOMException.ABORT_ERR
       context.error = response
-
-      observable.notify(context)
     } else if ('status' in response) {
       context.response = response
       context.responseType = response.type
       context.status = response.status
       context.isAborted = false
+    }
+    observable.notify(context)
+  }
 
-      observable.notify(context)
+  const onFulfilled = (response: Response) => {
+    const context = startContext as unknown as FetchCompleteContext
+    context.state = 'complete'
+    context.resolveDuration = elapsed(context.startClocks.timeStamp, timeStampNow())
+
+    const responseClone = response.clone()
+    const reader = responseClone.body?.getReader()
+
+    if (reader && ReadableStream) {
+      new ReadableStream({
+        start(controller) {
+          return pump()
+
+          function pump(): Promise<undefined> {
+            return (reader as ReadableStreamDefaultReader<Uint8Array>).read().then(({ done }) => {
+              if (done) {
+                controller.close()
+                context.duration = elapsed(context.startClocks.timeStamp, timeStampNow())
+                reportFetch(context, response)
+                return
+              }
+              return pump()
+            })
+          }
+        },
+      })
+    } else {
+      context.duration = elapsed(context.startClocks.timeStamp, timeStampNow())
+      reportFetch(context, response)
     }
   }
```

Comment:
💬 **suggestion**: ‏extract this logic in a function that you would use like this:

```ts
waitForResponseToFinish(response).then(() => {
   context.duration = elapsed(context.startClocks.timeStamp, timeStampNow())
   reportFetch(context, response)
})
```

<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏in a previous PR, I remember you used to factorize this code with the [logs network error collection](https://github.com/DataDog/browser-sdk/blob/0ab07cb70ac32cdd92677e80dd2942e7770e2cb9/packages/logs/src/domain/logsCollection/networkError/networkErrorCollection.ts#L142). Would it possible to do the same here?